### PR TITLE
CR & PR Guide: Integration branches

### PR DESCRIPTION
**Pull request by ulaulala**
Jira: -

**Description**
@mateuszkocz waiting for a decision about a way of creating protected branches

<details>
<summary><b>Screenshots</b></summary>
// put images here
</details>

**Self check**

- [x] Self CR'd
- [ ] Tests included
- [x] Description and screenshots attached

**PR Status**

- [ ] Code Review
- [ ] Quality Assurance

**Blockers**

**Deploy Notes**

### CODE REVIEW

File: src/content/code-review.mdx [hunk 1]
```diff
@@ -148,3 +148,34 @@ If it’s helpful for you - yes, you should. In some cases it’s easier to add
 
 :warning: The committed suggestion should be always validated if it doesn’t cause any issue, (e.g. fails of type checking). Take care of that.
 
+## Can I use integration branches?
+No, you should avoid integration branches. It's because they usually cause really big merges to a master branch, which can be dangerous even if every part of code was reviewed and tested earlier.
+Generally the best practise is to merge as small part of code as it's possible.
```
Comment (mateuszkocz) [inline src/content/code-review.mdx:153]: As soon as it’s possible

Comment (ulaulala) [reply src/content/code-review.mdx:153]: I mean as small as it's possible 🤔

Comment (mateuszkocz) [reply src/content/code-review.mdx:153]: Sorry, misread that.

**Comment by ulaulala** [comment]
@mateuszkocz [line 170](https://github.com/Appnroll/appnroll-developers/pull/30/files/f00c6c26d291c48b6bd5632c0da9e4edcdea819f#diff-b9bb5dc7ea1864a001b2b9513fe839b4R170) is about feature-flags 🤔 It's included in the first method as it's part

File: src/content/code-review.mdx [hunk 2]
```diff
@@ -148,3 +148,34 @@ If it’s helpful for you - yes, you should. In some cases it’s easier to add
...
+
+It may be not clear what to do with large tasks, which should be presented on production only whole. There are more than one solution for these kind of situations:
+
+**Puzzle / Chain**
+
+If it's possible a large task should be split into smaller parts and then these parts should represents parts of code (PRs). If you need to create many components to implement something, then use storybook to implement each component separately [1].
+This way the component is able to review (small PR) and test (visible in storybook).
```
Comment (mateuszkocz) [inline src/content/code-review.mdx:160]:
```suggestion
This way the component is able to be reviewed (small PR) and tested (visible in storybook).
```

**Comment by mateuszkocz** [comment]
> @mateuszkocz [line 170](https://github.com/Appnroll/appnroll-developers/pull/30/files/f00c6c26d291c48b6bd5632c0da9e4edcdea819f#diff-b9bb5dc7ea1864a001b2b9513fe839b4R170) is about feature-flags 🤔 It's included in the first method as it's part

Feature flags are not part of any puzzle or chain. It's an entirely different way of developing large chunks of functionalities that allow the usual process while maintaining the separation of production and staging environments.

**Pull request merged**